LibreSSL does not have some of the APIs that were added in OpenSSL 1.1.0 so we
need this.
--- include/libssl_compat.h	Tue Mar 21 15:04:20 2017
+++ include/libssl_compat.h	Tue Jul  4 20:56:45 2017
@@ -37,7 +37,7 @@
 #endif
 
 /* ----------------------------------------------------------------- */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 /* ----------------------------------------------------------------- */
 
 # include <openssl/objects.h>
--- libntp/libssl_compat.c	Wed Feb  1 11:47:13 2017
+++ libntp/libssl_compat.c	Tue Jul  4 21:13:13 2017
@@ -26,7 +26,7 @@
 /* ----------------------------------------------------------------- */
 
 /* ----------------------------------------------------------------- */
-#if defined(OPENSSL) && OPENSSL_VERSION_NUMBER < 0x10100000L
+#if defined(OPENSSL) && (OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER))
 /* ----------------------------------------------------------------- */
 
 #include "libssl_compat.h"
